airflow/providers/amazon/aws/hooks/sagemaker_notebook.py

```diff
@@ -0,0 +1,153 @@
+        :param config: Additional configuration options for the create call.
+        :param aws_conn_id: The AWS connection ID to use.
+
+        :return: A dict containing the information about the create notebook instance call.
+        """
+        if tags is not None:
+            tags = format_tags(tags)
+
+        create_notebook_instance_kwargs = {
+            "NotebookInstanceName": instance_name,
+            "InstanceType": instance_type,
+            "RoleArn": role_arn,
+            "Tags": tags,
+            "VolumeSizeInGB": volume_size_in_gb,
+            "KmsKeyId": volume_kms_key_id,
+            "SubnetId": subnet_id,
+            "SecurityGroupIds": security_group_ids,
+            "LifecycleConfigName": lifecycle_config_name,
+            "DirectInternetAccess": direct_internet_access,
+            "AcceleratorTypes": accelerator_types,
+            "DefaultCodeRepository": default_code_repo,
+            "AdditionalCodeRepositories": additional_code_repos,
+            "RootAccess": root_access,
+            "PlatformIdentifier": platform_id,
+        }
+
+        print(f"trimmed values are: {trim_none_values(create_notebook_instance_kwargs)}")
+        test = trim_none_values(create_notebook_instance_kwargs)
+        resp = self.conn.create_notebook_instance(**test, **kwargs)
+        return resp
+
+    def start_instance(self, *, instance_name: str, **kwargs: Any) -> None:
+        """Start a notebook instance.
+
+        :param instance_name: The name of the notebook instance.
+        :param wait_for_completion: Whether to wait for the start to complete.
+        :param kwargs: Additional arguments for the start notebook instance call.
+        :return: A dict containing the information about the start notebook instance call.
+        """
+        resp = self.conn.start_notebook_instance(NotebookInstanceName=instance_name, **kwargs)
+
+        return resp
+
+    def stop_instance(self, *, instance_name: str, **kwargs: Any) -> None:
+        """Stop a notebook instance.
+
+        :param instance_name: The name of the notebook instance.
+        :param wait_for_completion: Whether to wait for the stop to complete.
+        :param kwargs: Additional arguments for the stop notebook instance call.
+        :return: A dict containing the information about the stop notebook instance call.
+        """
+        resp = self.conn.stop_notebook_instance(NotebookInstanceName=instance_name, **kwargs)
+
+        return resp
+
+    def delete_instance(self, *, instance_name: str, **kwargs: Any) -> None:
+        """Delete a notebook instance.
+
+        :param instance_name: The name of the notebook instance.
+        :param kwargs: Additional arguments for the delete notebook instance call.
+        :return: A dict containing the information about the delete notebook instance call.
+        """
+        resp = self.conn.delete_notebook_instance(NotebookInstanceName=instance_name, **kwargs)
+
+        return resp
```

Comment:
Same here

Reply:
Refactored the hook operator to implement this suggestion.